src/test/java/uk/gov/objecthash/ObjectHashTest.java

```diff
@@ -48,4 +50,17 @@ public void hashesDict() {
         String digest = objectHash.hexDigest(data);
         assertEquals(digest, "7ef5237c3027d6c58100afadf37796b3d351025cf28038280147d42fdc53b960");
     }
+
+    @Test
+    public void returnsSuppliedHashIfRedacted() {
+        StringValue original = new StringValue("hello");
+        
+        assertEquals(original.hexDigest(), original.redacted().hexDigest());
+    }
+
+    @Test
+    public void hashesKnownRedactedHash() {
```

Comment:
could we test that an invalid one throws the custom exception?

Reply:
Yes, although I think actually it makes more sense to not throw the exception and hash as a string in this case. I will see how its handled in the original library